pilot/pkg/proxy/envoy/watcher.go

```diff
@@ -75,6 +72,29 @@ func (w *watcher) Run(ctx context.Context) {
 
 	go watchCerts(ctx, certDirs, watchFileEvents, defaultMinDelay, w.SendConfig)
 
+	shouldKickStart := true
+	if w.waitForCerts {
+		// If none of the cert/key files exist, should not kick start.
+		shouldKickStart = false
+		for _, cert := range w.certs {
+			for _, filename := range cert.Files {
+				if _, err := os.Stat(path.Join(cert.Directory, filename)); err == nil || !os.IsNotExist(err) {
+					shouldKickStart = true
+					break
```

Comment:
Consider using https://golang.org/ref/spec#Break_statements to exit the entire loop when a kickstart is required.

Reply:
Done. Thanks.